pkg/operator/ceph/csi/secrets_test.go

```diff
@@ -124,3 +136,161 @@ func Test_deleteOwnedCSISecretsByCephCluster(t *testing.T) {
 		t.Errorf("expected unowned secret %q to still exist", CsiCephFSProvisionerSecret)
 	}
 }
+
+func TestSortCsiClientName(t *testing.T) {
+	tests := []struct {
+		name    string
+		args    []string
+		want    []string
+		wantErr bool
+	}{
+		{
+			name: "testName",
+			args: keys,
+			want: []string{
+				"client.csi-rbd-node", "client.csi-rbd-node.1", "client.csi-rbd-node.2", "client.csi-rbd-node.4", "client.csi-rbd-node.10",
+			},
+			wantErr: false,
+		},
```

Comment:
More tests!

The usual test cases I ensure I write for anything that takes a list:
- list is empty
- list has one element
- list has 2+ elements

For this, I would also suggest sorting a list that has elements with different base names like `client.csi-cephfs-node`, `client.csi-rbd-node`.

And also sort those with number suffixes added like `client.csi-cephfs-node`, `client.csi-rbd-node`, `client.csi-cephfs-node.1`, `client.csi-rbd-node.1`.

Reply:
updated